card/lib/card/view_name.rb

```diff
@@ -0,0 +1,44 @@
+# -*- encoding : utf-8 -*-
+require_dependency 'card/env'
+
+require 'smart_name'
+
+class Card
+  class ViewName < SmartName
+    @@name2viewnameobject = {}
+
+    class << self
+      def new obj
+        return obj if self.class===obj
+        str = Array===obj ? obj*joint : obj.to_s
+        if known_name = @@name2viewnameobject[str]
+          known_name
+        else
+          super str.strip
+        end
+      end
+    end
+
+    def initialize str
+      @s = str.to_s.strip
+      @s = @s.encode('UTF-8') if RUBYENCODING
+      @key = if @s.index(self.class.joint)
+          @parts = @s.split(/\s*#{JOINT_RE}\s*/)
+          @parts << '' if @s[-1,1] == self.class.joint
+          @simple = false
+          @parts.map { |p| p.to_name.key } * self.class.joint
+        else
+          @parts = [str]
+          @simple = true
+          str.empty? ? '' : simple_key
+        end
+      @@name2viewnameobject[str] = self
+    end
+
+    def simple_key
+      decoded.underscore.gsub(/[^#{OK4KEY_RE}]+/,'_').split(/_+/).reject(&:empty?)*'_'
+    end
```

Comment:
Whoa, that's a great point.  If we do eventually have view cards, then they would be mapped to coded views via codenames.  So, **there's no great reason to support variable view names in code references to views**, is there? It's just extra processing.

The only time you'd really want to do view name processing is in interpreting views as specified in content (nest syntax). And for now, we can probably get away without supporting view name variants in nests.

Fwiw, we do still need to handle (even today) the old aliased/deprecated names (eg "core" used to be "naked" and is still probably used in some old sites).  But I'm not sure there's much need for name objects without supporting name variants, is there?